src/formatDate.js

```diff
@@ -50,7 +50,51 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromDateSeparator = fromFormat[3];
+  const toDateSeparator = toFormat[3];
+  const yearsMap = {};
+  let formatMap = {};
+  const dateParts = date.split(fromDateSeparator);
+  const formatedDate = [];
+
+  const getFullYear = (year) => {
+    if (+year < 30) {
+      return `20${year}`;
+    }
+
+    return `19${year}`;
+  };
+
+  const getShortYear = (year) => {
+    return year.slice(2);
+  };
+
```

Comment:
It would be better to take these functions out from formatDate function so that they are not created every time the formatDate is called.
```suggestion
```